game-core/src/main/java/games/strategy/triplea/delegate/battle/TargetGroup.java

```diff
@@ -0,0 +1,94 @@
+package games.strategy.triplea.delegate.battle;
+
+import com.google.common.collect.Sets;
+import games.strategy.engine.data.GamePlayer;
+import games.strategy.engine.data.Unit;
+import games.strategy.engine.data.UnitType;
+import games.strategy.triplea.attachments.UnitAttachment;
+import games.strategy.triplea.delegate.Matches;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Comparator;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+import lombok.Getter;
+import lombok.Setter;
+import org.triplea.java.collections.CollectionUtils;
+
+/** Group of firing units and their targets. */
+@Getter
```

Comment:
It looks like the getter here is used only internally. Unless I missed a usage, do you have any thoughts on removing the getter or restricting the visibility of the get method? 

Looking at the class, and the class properties, it raises a code smell if we return the internal data, it makes it seem as if the class were incompletely providing operations on the data members. IE: if this class provides operations on two internal sets, then what's the point to return the internal data unless this class is incomplete in doing so.

Reply:
The getters are used in MustFightBattle to pull the information out.

Reply:
I see that `getFiringUnits` and `getTargetUnits` are used in `MustFightBattle`, though the only usages of the generated getters are in `TargetGroup`. Can you point me to where they are in `MustFightBattle`, am I looking at the wrong diff or something?

Reply:
Ah I see. Yeah, I reduced the visibility of the getters.

Reply:
Personally I would skip the generated getters altogether for more directness. It also makes it clear at the call site what we are doing and a reader would not have to verify where the getter is and if it could be overloaded.

Good enough though marking the getter private to reduce its scope. :+1: